tensilelite/Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp

```diff
@@ -845,6 +845,64 @@ namespace Tensile
                 }
             };
 
+            struct AmaxDStatusCheck
+                : public Predicate_CRTP<AmaxDStatusCheck, ContractionProblemGemm>
+            {
+                enum
+                {
+                    HasIndex = false,
+                    HasValue = true
+                };
+                bool value;
+
+                AmaxDStatusCheck() = default;
+                AmaxDStatusCheck(bool value)
+                    : value(value)
+                {
+                }
+
+                static std::string Type()
+                {
+                    return "AmaxDStatusCheck";
+                }
+
+                virtual bool operator()(ContractionProblemGemm const& problem) const override
+                {
+                    bool amaxDStatusEqual = (problem.outputAmaxD() == value);
```

Comment:
Doen't reject for case "!problem.outputAmaxD && value"

Reply:
I guess this is the same question as you asked: can AmaxDS support AmaxD = nullptr ?
Unfortunately no in this version. I was not having the idea to make it kind of universal. So that's why currently we need AmaxD kernels and non-AmaxD kernels. And using this predicate to make sure if this kernel is non-AmaxD / AmaxD one. 

I'm afraid that make it universal needs more time. And also, AmaxD kernels insert one extra fabs instruction before result *= ScaleD, and is scattering everywhere in GlobalWriteBatch. If we want to make it universal one, does it mean we need to branch before each extra fabs instruction ? That's what I'm concerning now. (For the final reduction part, branch to skip it is easy.)